activerecord/lib/active_record/counter_cache.rb

```diff
@@ -112,13 +150,30 @@ def increment_counter(counter_name, id)
       #
       # * +counter_name+ - The name of the field that should be decremented.
       # * +id+ - The id of the object that should be decremented or an array of ids.
+      # * <tt>:touch</tt> - Touch timestamp columns when updating.
+      #   Pass +true+ to touch +updated_at+ and/or +updated_on+. Pass a symbol to
+      #   touch that column or an array of symbols to touch just those ones.
       #
       # ==== Examples
       #
       #   # Decrement the posts_count column for the record with an id of 5
       #   DiscussionBoard.decrement_counter(:posts_count, 5)
-      def decrement_counter(counter_name, id)
-        update_counters(id, counter_name => -1)
+      #
+      #   # Decrement the posts_count column for the record with an id of 5
+      #   # and update the updated_at value.
+      #   DiscussionBoard.decrement_counter(:posts_count, 5, touch: true)
+      def decrement_counter(counter_name, id, touch: nil)
+        update_counters(id, { counter_name => -1 }.merge(touch: touch))
+      end
+
+      private
+
+      def resolve_timestamp_columns(object, touch)
+        if touch == true
+          object.send(:timestamp_attributes_for_update_in_model)
+        else
+          Array(touch)
```

Comment:
When `touch` is `false` it is trying to update it as a column. We should handle this case properly.